### ISSUE TITLE
7月份拟迁移至module的开放课题视觉算法

### ISSUE THREAD

**Issue by JiaojiaoYe1994** [Oneflow-Inc/models#31]
将oneflow实现的视觉算法迁移至oneflow新接口。Oneflow新接口和PyTorch module完全对齐。

### 一、视觉算法选择

浙工大拟迁移到module的算法

<!DOCTYPE html><figure class="md-table-fig" cid="n29" mdtype="table" style="box-sizing: border-box; margin: 1.2em 0px; overflow-x: auto; max-width: calc(100% + 16px); padding: 0px; cursor: default; color: rgb(31, 9, 9); font-family: &quot;PT Serif&quot;, &quot;Times New Roman&quot;, Times, serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-style: initial; text-decoration-color: initial;">

MODEL | 负责人 | 缺失算子 | pr
-- | -- | -- | --
PoseNet |   |   |   
SCNet |   |   | 

</figure><br class="Apple-interchange-newline">

[算法库链接](https://github.com/Oneflow-Inc/oneflow_vision_model) , 建议对照pytorch代码迁移算法，而不是oneflow旧接口代码。

### 二、迁移Example

参考算法　[ResNet50](https://github.com/Oneflow-Inc/models/tree/main/resnet50)

```
# pytorch module
import torch
import torch.nn as nn

# oneflow module
# oneflow新接口统一都在oneflow.experimental.nn 下面
# import oneflow.experimental as torch
# import oneflow.experimental.nn as nn
import oneflow.experimental as flow
import oneflow.experimental.nn as nn
```

### 三、模型验收标准

复现好的模型，代码提交 [Oneflow-Inc/models](https://github.com/Oneflow-Inc/models) 仓库统一管理。

对于能找到Pytorch官方实现的模型，比如视觉分类网络，直接对齐torch实现然后把torch的预训练模型直接转成oneflow来用，需要提供 训练脚本、预训练模型、预测脚本、 requirements.txt 和README。

获取运行当前目录下 python 脚本所需的 requirements.txt

```python3
pip3 install pipreqs --user
pipreqs .
```

然后把生成的 requirements.txt 里面库的版本号去掉

README 的内容和格式参考：[resnet50](https://github.com/Oneflow-Inc/models/blob/main/resnet50/README.md)，如果是生成类或者像素级任务，比如 GAN，风格化、分割和超分等等，则需要在 README.md 中提供可视化的结果。

预训练模型需要上传到阿里云，可以暂时先发给我统一上传。模型保存方式，可以参考：
https://github.com/Oneflow-Inc/models/blob/main/resnet50/train_oneflow.py#L120

预测或者训练python脚本需同时提供 bash 脚本，用于方便用户设置一些可变的命令行参数。比如预测脚本，可在bash脚本中下载预训练模型，尽量要让用户下载代码装好oneflow之后，直接运行 bash 脚本就能跑起来。比如 [resnet50/infer.sh](https://github.com/Oneflow-Inc/models/blob/main/resnet50/infer.sh)

### 四、开发参考

  * model 合并分支: https://github.com/Oneflow-Inc/models/

  * 算子统计和进度汇报统计: https://github.com/Oneflow-Inc/models/issues/31

  * oneflow lazy 版本实现：https://github.com/Oneflow-Inc/oneflow_vision_model

  * oneflow API文档：https://oneflow.readthedocs.io/en/master/

**Comment by XinYangDong** [comment]
> 将oneflow实现的视觉算法迁移至oneflow新接口。Oneflow新接口和PyTorch module完全对齐。
> 
> ### 一、视觉算法选择
> 浙工大拟迁移到module的算法
> 
> MODEL	负责人	缺失算子	pr
> PoseNet	 		 
> SCNet	 	 董信阳	无缺失     无
> [算法库链接](https://github.com/Oneflow-Inc/oneflow_vision_model) , 建议对照pytorch代码迁移算法，而不是oneflow旧接口代码。
> 
> ### 二、迁移Example
> 参考算法　[ResNet50](https://github.com/Oneflow-Inc/models/tree/main/resnet50)
> 
> ```
> # pytorch module
> import torch
> import torch.nn as nn
> 
> # oneflow module
> # oneflow新接口统一都在oneflow.experimental.nn 下面
> # import oneflow.experimental as torch
> # import oneflow.experimental.nn as nn
> import oneflow.experimental as flow
> import oneflow.experimental.nn as nn
> ```
> 
> ### 三、模型验收标准
> 复现好的模型，代码提交 [Oneflow-Inc/models](https://github.com/Oneflow-Inc/models) 仓库统一管理。
> 
> 对于能找到Pytorch官方实现的模型，比如视觉分类网络，直接对齐torch实现然后把torch的预训练模型直接转成oneflow来用，需要提供 训练脚本、预训练模型、预测脚本、 requirements.txt 和README。
> 
> 获取运行当前目录下 python 脚本所需的 requirements.txt
> 
> ```python
> pip3 install pipreqs --user
> pipreqs .
> ```
> 
> 然后把生成的 requirements.txt 里面库的版本号去掉
> 
> README 的内容和格式参考：[resnet50](https://github.com/Oneflow-Inc/models/blob/main/resnet50/README.md)，如果是生成类或者像素级任务，比如 GAN，风格化、分割和超分等等，则需要在 README.md 中提供可视化的结果。
> 
> 预训练模型需要上传到阿里云，可以暂时先发给我统一上传。模型保存方式，可以参考：
> https://github.com/Oneflow-Inc/models/blob/main/resnet50/train_oneflow.py#L120
> 
> 预测或者训练python脚本需同时提供 bash 脚本，用于方便用户设置一些可变的命令行参数。比如预测脚本，可在bash脚本中下载预训练模型，尽量要让用户下载代码装好oneflow之后，直接运行 bash 脚本就能跑起来。比如 [resnet50/infer.sh](https://github.com/Oneflow-Inc/models/blob/main/resnet50/infer.sh)
> 
> ### 四、开发参考
> * model 合并分支: https://github.com/Oneflow-Inc/models/
> * 算子统计和进度汇报统计: [7月份拟迁移至module的开放课题视觉算法 #31](https://github.com/Oneflow-Inc/models/issues/31)
> * oneflow lazy 版本实现：https://github.com/Oneflow-Inc/oneflow_vision_model
> * oneflow API文档：https://oneflow.readthedocs.io/en/master/

**Comment by gemessss** [comment]
> 将oneflow实现的视觉算法迁移至oneflow新接口。Oneflow新接口和PyTorch module完全对齐。
> 
> ### 一、视觉算法选择
> 浙工大拟迁移到module的算法
> 
> MODEL	负责人	缺失算子	pr
> PoseNet	葛逸凡		 
> SCNet	 	 	
> [算法库链接](https://github.com/Oneflow-Inc/oneflow_vision_model) , 建议对照pytorch代码迁移算法，而不是oneflow旧接口代码。
> 
> ### 二、迁移Example
> 参考算法　[ResNet50](https://github.com/Oneflow-Inc/models/tree/main/resnet50)
> 
> ```
> # pytorch module
> import torch
> import torch.nn as nn
> 
> # oneflow module
> # oneflow新接口统一都在oneflow.experimental.nn 下面
> # import oneflow.experimental as torch
> # import oneflow.experimental.nn as nn
> import oneflow.experimental as flow
> import oneflow.experimental.nn as nn
> ```
> 
> ### 三、模型验收标准
> 复现好的模型，代码提交 [Oneflow-Inc/models](https://github.com/Oneflow-Inc/models) 仓库统一管理。
> 
> 对于能找到Pytorch官方实现的模型，比如视觉分类网络，直接对齐torch实现然后把torch的预训练模型直接转成oneflow来用，需要提供 训练脚本、预训练模型、预测脚本、 requirements.txt 和README。
> 
> 获取运行当前目录下 python 脚本所需的 requirements.txt
> 
> ```python
> pip3 install pipreqs --user
> pipreqs .
> ```
> 
> 然后把生成的 requirements.txt 里面库的版本号去掉
> 
> README 的内容和格式参考：[resnet50](https://github.com/Oneflow-Inc/models/blob/main/resnet50/README.md)，如果是生成类或者像素级任务，比如 GAN，风格化、分割和超分等等，则需要在 README.md 中提供可视化的结果。
> 
> 预训练模型需要上传到阿里云，可以暂时先发给我统一上传。模型保存方式，可以参考：
> https://github.com/Oneflow-Inc/models/blob/main/resnet50/train_oneflow.py#L120
> 
> 预测或者训练python脚本需同时提供 bash 脚本，用于方便用户设置一些可变的命令行参数。比如预测脚本，可在bash脚本中下载预训练模型，尽量要让用户下载代码装好oneflow之后，直接运行 bash 脚本就能跑起来。比如 [resnet50/infer.sh](https://github.com/Oneflow-Inc/models/blob/main/resnet50/infer.sh)
> 
> ### 四、开发参考
> * model 合并分支: https://github.com/Oneflow-Inc/models/
> * 算子统计和进度汇报统计: [7月份拟迁移至module的开放课题视觉算法 #31](https://github.com/Oneflow-Inc/models/issues/31)
> * oneflow lazy 版本实现：https://github.com/Oneflow-Inc/oneflow_vision_model
> * oneflow API文档：https://oneflow.readthedocs.io/en/master/

**Comment by gemessss** [comment]
> > 将oneflow实现的视觉算法迁移至oneflow新接口。Oneflow新接口和PyTorch module完全对齐。
> > ### 一、视觉算法选择
> > 浙工大拟迁移到module的算法
> > MODEL	负责人	缺失算子	pr
> > PoseNet	葛逸凡	无缺失	lnception-ResnetV2
> > SCNet	 	 董信阳	无缺失     无
> > [算法库链接](https://github.com/Oneflow-Inc/oneflow_vision_model) , 建议对照pytorch代码迁移算法，而不是oneflow旧接口代码。
> > ### 二、迁移Example
> > 参考算法　[ResNet50](https://github.com/Oneflow-Inc/models/tree/main/resnet50)
> > ```
> > # pytorch module
> > import torch
> > import torch.nn as nn
> > 
> > # oneflow module
> > # oneflow新接口统一都在oneflow.experimental.nn 下面
> > # import oneflow.experimental as torch
> > # import oneflow.experimental.nn as nn
> > import oneflow.experimental as flow
> > import oneflow.experimental.nn as nn
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ### 三、模型验收标准
> > 复现好的模型，代码提交 [Oneflow-Inc/models](https://github.com/Oneflow-Inc/models) 仓库统一管理。
> > 对于能找到Pytorch官方实现的模型，比如视觉分类网络，直接对齐torch实现然后把torch的预训练模型直接转成oneflow来用，需要提供 训练脚本、预训练模型、预测脚本、 requirements.txt 和README。
> > 获取运行当前目录下 python 脚本所需的 requirements.txt
> > ```python
> > pip3 install pipreqs --user
> > pipreqs .
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 然后把生成的 requirements.txt 里面库的版本号去掉
> > README 的内容和格式参考：[resnet50](https://github.com/Oneflow-Inc/models/blob/main/resnet50/README.md)，如果是生成类或者像素级任务，比如 GAN，风格化、分割和超分等等，则需要在 README.md 中提供可视化的结果。
> > 预训练模型需要上传到阿里云，可以暂时先发给我统一上传。模型保存方式，可以参考：
> > https://github.com/Oneflow-Inc/models/blob/main/resnet50/train_oneflow.py#L120
> > 预测或者训练python脚本需同时提供 bash 脚本，用于方便用户设置一些可变的命令行参数。比如预测脚本，可在bash脚本中下载预训练模型，尽量要让用户下载代码装好oneflow之后，直接运行 bash 脚本就能跑起来。比如 [resnet50/infer.sh](https://github.com/Oneflow-Inc/models/blob/main/resnet50/infer.sh)
> > ### 四、开发参考
> > 
> > * model 合并分支: https://github.com/Oneflow-Inc/models/
> > * 算子统计和进度汇报统计: [7月份拟迁移至module的开放课题视觉算法 #31](https://github.com/Oneflow-Inc/models/issues/31)
> > * oneflow lazy 版本实现：https://github.com/Oneflow-Inc/oneflow_vision_model
> > * oneflow API文档：https://oneflow.readthedocs.io/en/master/

**Comment by 15168387900** [comment]
TTSR所需的算子：

torch.from_numpy
torch.utils.data.Dataset
torchvision.transforms.Compose
torch.utils.data.DataLoader
torch.device
torch.rand
torch.autograd.grad
torch.nn.DataParallel
torch.nn.functional.mse_loss
torch.nn.functional.interpolate
torch.nn.functional.l1_loss
torchvision.models.vgg19
torch.nn.functional.relu
torch.bmm
torch.nn.functional.unfold
torch.nn.functional.normalize
torch.nn.functional.fold
torch.load
torch.no_grad
torch.save
torch.Tensor
torch.eye
torch.view
